core/domain/platform_parameter_registry_test.py

```diff
@@ -0,0 +1,314 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the platform parameter registry."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import platform_parameter_domain as parameter_domain
+from core.domain import platform_parameter_registry as registry
+from core.platform import models
+from core.tests import test_utils
+import feconf
+import utils
+
+
+(config_models,) = models.Registry.import_models(
+    [models.NAMES.config])
+memcache_services = models.Registry.import_memcache_services()
+
+
+class PlatformParameterRegistryTests(test_utils.GenericTestBase):
+    """Tests for the platform parameter Registry."""
+
+    def setUp(self):
+        super(PlatformParameterRegistryTests, self).setUp()
+
+        self.original_param_registry = registry.Registry.parameter_registry
+        registry.Registry.parameter_registry.clear()
+
+        # Parameter names that might be used in following tests.
+        parameter_names = ('parameter_a', 'parameter_b')
+        memcache_keys = [
+            parameter_domain.PlatformParameter.get_memcache_key(name)
+            for name in parameter_names]
+        memcache_services.delete_multi(memcache_keys)
+
+    def tearDown(self):
+        super(PlatformParameterRegistryTests, self).tearDown()
+
+        registry.Registry.parameter_registry = self.original_param_registry
+
+    def create_example_parameter_with_name(self, name):
```

Comment:
What is this test verifying? Per https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests test names should have both the conditions being tested and the expected outcome of the action under these conditions. In this case the action & conditions are clear, but what do we actually expect to happen? Nothing seems to be verified in the test.

Ditto for all other tests in this PR, where relevant (please make sure all of your tests have clear conditions & outcomes in their names, and in the tests themselves).

Reply:
This method is a utility method that creates parameter with given name and some example rules, not a test case.
Took a pass on the other test cases, renamed some of them for clarity of the testing purposes.